crypto/certificate.go

```diff
@@ -0,0 +1,114 @@
+package crypto
+
+import (
+	"errors"
+)
+
+// Signable defines an interface which would generate a unique hash based on its content
+type Signable interface {
+	Hash() []byte
+}
+
+// PublicKey defines an interface of Public Key
+type PublicKey interface {
+	Bytes() []byte
+	Str() string
+}
+
+// Signature defines an interface of
+type Signature interface {
+	Bytes() []byte
+}
+
+// Verifier defines an interface which could verify a signature
+type Verifier interface {
+	Verify(signature Signature) bool
+	PublicKey() PublicKey
+}
+
+// Signer defines an interface which could sign a signable
+type Signer interface {
+	Verifier
+	Sign(Signable) (Signature, error)
+}
+
+// Certificate represents a certificate for a signable
+type Certificate struct {
```

Comment:
Is it better one sig/pk per Certificate?

Reply:
In most cases, we will store a list of signatures for a given signable, thus the signable will be redundant data if we store one sig/pk per certificate.